tests/test_requests.py

```diff
@@ -240,11 +240,32 @@ def test_request_is_disconnected():
     assert disconnected_after_response
 
 
+def test_request_state_object():
+    scope = {'state': {'old': 'foo'}}
+
+    s = State(scope['state'])
+    assert s._state == scope['state']
+    assert getattr(s, '_state') == scope['state']
+
+    s.new = 'value'                   # test __setattr__
+    assert s.new == 'value'           # test __getattr__
+    assert s._state['new'] == 'value' # test if inner _state dict is updated.
+
+    del s.new # test __delattr__
+
+    try:
+        assert s.new == 'value' # will bombed with AttributeError
+    except AttributeError as e:
+        assert str(e) == "'State' object has no attribute 'new'"
+
+    del s._state # Deleting _state dict should not bombed.
```

Comment:
We won't need this line, once we've dropped the `__delattr__` case for `_state`.